packages/actions-server/src/ActionsProxy.test.ts

```diff
@@ -0,0 +1,279 @@
+import { describe, it, expect, beforeEach, afterEach } from "vitest";
+import { ActionsProxy } from "./ActionsProxy.js";
+import { Client as McpClient } from "@modelcontextprotocol/sdk/client/index.js";
+import { InMemoryTransport } from "@modelcontextprotocol/sdk/inMemory.js";
+import { z } from "zod";
+import { WebSocket } from "ws";
+
+interface ToolResult {
+    content: Array<{
+        type: "text";
+        text: string;
+    }>;
+    _meta: Record<string, unknown>;
+    isError: boolean;
+}
+
+interface ToolRequest {
+    type: 'TOOL_REQUEST';
+    payload: {
+        toolName: string;
+        parameters: Record<string, unknown>;
+        requestId: string;
+    };
+}
```

Comment:
these types will be moved to a proper place once – or shall be replaced by MCP native types